src/System.Device.Gpio/System/Device/Gpio/Drivers/RaspberryPi3Driver.cs

```diff
@@ -22,6 +22,23 @@ private void ValidatePinNumber(int pinNumber)
             }
         }
 
+        private void ValidatePinIsInput(int pinNumber)

```

Comment:
If this isn't really used in the Windows implementation and is private, I would keep it in the .Linux specific implementation.

Reply:
Renaming the method to ValidatePinIsOpen and moving the the .linux part. Then that's what should actually be tested, I believe.

Reply:
Note: The Windows implementation already behaves that way: Any operation on a pin that is not open throws, except for opening it and for closing it (probably because repeatedly closing a pin should just be ignored)